src/main/java/core/basesyntax/MyTask.java

```diff
@@ -1,19 +1,73 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.concurrent.RecursiveTask;
 
 public class MyTask extends RecursiveTask<Long> {
+    private static int maxValue;
     private int startPoint;
     private int finishPoint;
 
     public MyTask(int startPoint, int finishPoint) {
         this.startPoint = startPoint;
         this.finishPoint = finishPoint;
+        if (Thread.currentThread().getName().equals("main")) {
+            maxValue = Math.max(startPoint,finishPoint);
+        }
+
     }
 
     @Override
     protected Long compute() {
-        // write your code here
-        return null;
+        if (Math.max(Math.abs(finishPoint),Math.abs(startPoint))
+                - Math.min(Math.abs(finishPoint),Math.abs(startPoint)) > 10) {
+            System.out.println(Thread.currentThread().getName()
+                    + " Splitting task "
+                    + startPoint
+                    + " , finishPoint: "
+                    + finishPoint);
+            List<RecursiveTask<Long>> subTasks = new ArrayList<>(createSubTasks());
+            for (RecursiveTask<Long> task: subTasks) {
+                task.fork();
+            }
+            long result = 0;
+            for (RecursiveTask<Long> task: subTasks) {
+                result += task.join();
+            }
+            return result;
+        } else {
+            System.out.println("RecursiveAction: Doing myself, startPoint : "
+                    + startPoint
+                    + " , finishPoint: "
+                    + finishPoint
+                    + " maxValue: "
+                    + maxValue);
+            int value = 0;
+            if (finishPoint < startPoint) {
+                return (long) value;
+            } else if (finishPoint == maxValue && (finishPoint - startPoint) <= 10) {
+                value = (startPoint + finishPoint - 1) * (finishPoint - startPoint) / 2;
+            } else {
+                value = (startPoint + finishPoint) * (finishPoint - startPoint + 1) / 2;
+            }
+            return (long) value;
```

Comment:
Давай создадим отдельный метод calculateSum(), который будет суммировать все элементы от startPoint до finishPoint. И давай, для упрощения самой логики, будем считать, что startPoint и finishPoint всегда положительные и startPoint < finishPoint
```suggestion
           return calculateSum();
```